web/controllers/v1/user_controller.ex

```diff
@@ -56,4 +58,29 @@ defmodule Cog.V1.UserController do
     Repo.delete!(user)
     send_resp(conn, :no_content, "")
   end
+
+  ###############################################
+  # Plug function - local only to user controller
+  ###############################################
+  @spec check_self_updating(Plug.Conn.t, []) :: Plug.Conn.t
+  def check_self_updating(conn, opts \\ []) do
```

Comment:
Some documentation explaining the rationale for this would be useful.

Also, can this be private?